ctapipe/image/charge_extractors.py

```diff
@@ -105,589 +128,161 @@ def extract_charge(self, waveforms):
-        a window.
-
-        Attributes
-        ----------
-        neighbours : list
-            List of neighbours for each pixel. Changes per telescope.
-
-        Parameters
-        ----------
-        config : traitlets.loader.Config
-            Configuration specified by config file or cmdline arguments.
-            Used to set traitlet values.
-            Set to None if no configuration to pass.
-        tool : ctapipe.core.Tool
-            Tool executable that is calling this component.
-            Passes the correct logger to the component.
-            Set to None if no Tool to pass.
-        kwargs
-        """
-        super().__init__(config=config, parent=parent, **kwargs)
-
-    def get_peakpos(self, waveforms):
-        return self._obtain_peak_position(waveforms)
-
-    def _get_window_start(self, waveforms, peakpos):
-        return peakpos - self.window_shift
-
-    def _get_window_width(self, waveforms):
-        nchan, npix, nsamples = waveforms.shape
-        return np.full((nchan, npix), self.window_width, dtype=np.intp)
-
-    @abstractmethod
-    def _obtain_peak_position(self, waveforms):
-        """
-        Find the peak to define integration window around.
-
-        Parameters
-        ----------
-        waveforms : ndarray
-            Waveforms stored in a numpy array of shape
-            (n_chan, n_pix, n_samples).
-
-        Returns
-        -------
-        peakpos : ndarray
-            Numpy array of the peak position for each pixel. Has shape of
-            (n_chan, n_pix).
-
-        """
-
-
-class SimpleIntegrator(WindowIntegrator):
-    t0 = Int(0, help='Define the peak position for all '
-                     'pixels').tag(config=True)
-
-    def __init__(self, config=None, parent=None, **kwargs):
-        """
-        Charge extractor that integrates within a window defined by the user.
-
-        Attributes
-        ----------
-        neighbours : list
-            List of neighbours for each pixel. Changes per telescope.
-
-        Parameters
-        ----------
-        config : traitlets.loader.Config
-            Configuration specified by config file or cmdline arguments.
-            Used to set traitlet values.
-            Set to None if no configuration to pass.
-        tool : ctapipe.core.Tool
-            Tool executable that is calling this component.
-            Passes the correct logger to the component.
-            Set to None if no Tool to pass.
-        kwargs
-        """
-        super().__init__(config=config, parent=parent, **kwargs)
-
-    def _obtain_peak_position(self, waveforms):
-        nchan, npix, nsamples = waveforms.shape
-        return np.full((nchan, npix), self.t0, dtype=np.intp)
+        # TODO: remove integration window return
+        peakpos = np.zeros(waveforms.shape[:2], dtype=np.intp)
+        window = np.ones(waveforms.shape, dtype=np.bool)
+        return waveforms.sum(2), peakpos, window
 
 
-class PeakFindingIntegrator(WindowIntegrator):
```

Comment:
was `PeakFindingIntegrator` removed without replacement? If so, why?

Reply:
Ah I see, `PeakFindingIntegrator` was abstract and only used in order to derive form it, right?

Reply:
Correct